hyppo/independence/hhg.py

```diff
@@ -129,16 +176,35 @@ def statistic(self, x, y):
         distx = x
         disty = y
 
-        if not self.is_distance:
-            distx, disty = compute_dist(
-                x, y, metric=self.compute_distance, **self.kwargs
-            )
-
-        S = _pearson_stat(distx, disty)
-        mask = np.ones(S.shape, dtype=bool)
-        np.fill_diagonal(mask, 0)
-        stat = np.sum(S[mask])
-        self.stat = stat
+        if not self.auto:
+            if not self.is_distance:
+                distx, disty = compute_dist(
+                    x, y, metric=self.compute_distance, **self.kwargs
+                )
+            S = _pearson_stat(distx, disty)
+            mask = np.ones(S.shape, dtype=bool)
+            np.fill_diagonal(mask, 0)
+            stat = np.sum(S[mask])
+            self.stat = stat
+
+        else:
+            # Fast HHG
+            if not self.is_distance:
+                pointer = (np.mean(x, axis=0), np.mean(y, axis=0))
+                zx, zy = pointer
```

Comment:
you're doing a lot of the same commands to two inputs from 194-205. Can you make it into a helper function?

Reply:
Good point! I'll do that immediately.

Reply:
Helper function added as _centerpoint_dist().